Weather/Views/WeatherViewController.swift

```diff
@@ -21,20 +21,16 @@ class WeatherViewController: UIViewController {
         super.viewDidLoad()
         
         let weatherView = WeatherView()
-        self.timedForecastPresenter = TimedForecastPresenterImplementation()
-        self.weekForecastPresenter = WeekForecastPresenterImplementation()
-        
-        for index in 1...10 {
-            let timedForecast = TimedForecast(at: String(index) + " AM", is: index)
-            let weekForecast = WeekdayForecast(on: String(index) + "day", temperatureAtMidday: index, temperatureAtNight: index)
-            
-            timedForecastPresenter!.addItem(timedForecast)
-            weekForecastPresenter!.addItem(weekForecast)
-        }
+        timedForecastPresenter = TimedForecastPresenterImplementation()
+        weekForecastPresenter = WeekForecastPresenterImplementation()
         
         weatherView.timedForecastView.dataSource = self
         weatherView.timedForecastView.delegate = self
         weatherView.timedForecastView.register(TimedForecastCellView.self, forCellWithReuseIdentifier: "timedForecastCell")
+        
+        weatherView.weekForecastView.dataSource = self
```

Comment:
To clean the code and also prevent from populating it with new code as it tend so grow too fast i would suggest making datasource for this collection view a separate object called `DataSource` (add prefix for where this datasource belongs eg `WeatherCollectionViewDataSource`).

Reply:
The same applies to every other datasource you stumble upon, that is a good practice.